src/models/util/ParseCalChart3Utils.ts

```diff
@@ -0,0 +1,96 @@
+export function readFourCharCode(buffer: DataView, offset: number): string {
+  return String.fromCharCode(
+    buffer.getUint8(offset + 0),
+    buffer.getUint8(offset + 1),
+    buffer.getUint8(offset + 2),
+    buffer.getUint8(offset + 3)
+  );
+}
+
+export function readTwoCharCode(buffer: DataView, offset: number): string {
+  return String.fromCharCode(
+    buffer.getUint8(offset + 0),
+    buffer.getUint8(offset + 1)
+  );
+}
+
+export function readInt32(buffer: DataView, offset: number): number {
+  return buffer.getInt32(offset);
+}
+
+export function readInt16(buffer: DataView, offset: number): number {
+  return buffer.getInt16(offset);
+}
+
+export function readStringTillEnd(buffer: DataView, offset: number): string {
+  let retVal = '';
+  while (offset < buffer.byteLength) {
+    const value = buffer.getUint8(offset++);
+    if (!value) {
+      break;
+    }
+    retVal += String.fromCharCode(value);
+  }
+  if (offset !== buffer.byteLength) {
+    throw 'String not parsed correctly';
+  }
+  return retVal;
+}
+
+export function
+readArrayOfStringsTillEnd(buffer: DataView, offset: number): string[] {
+  const retVal: string[] = [];
+
+  let currentLabel = '';
+  while (offset < buffer.byteLength) {
+    const value = buffer.getUint8(offset++);
+    if (value) {
+      currentLabel += String.fromCharCode(value);
+    } else {
+      retVal.push(currentLabel);
+      currentLabel = '';
+    }
+  }
+  if (currentLabel !== '') {
+    // if we have a partial label we did not end with a '\0'.  Error.
+    throw 'Error parsing strings from block';
```

Comment:
For the thrown errors, we don't have to explicitly prefix the string with "error" since it's already known that it is an error. Please double check the other error messages as well.
```suggestion
    throw 'Label did not end with \\0';
```